### PR TITLE
UI Components Button: Fixing Disabled issue #26541

### PR DESCRIPTION
**Pull request by Amstutz**
PR for fixing [#26541](https://mantis.ilias.de/view.php?id=26541).

This PR contains:
- Necessary changes in rendering templates (getting rid of ilSubmitInactive/disabled, as proposed by @alex40724 )
- Changes in less, bootstrap, note disabled and ilSubmitInactive have been completely abandoned in the less for button in the UI Components 
- extended examples to show case and document the inactive state better.
- fix of existing tests

### CODE REVIEW

File: src/UI/templates/default/Button/tpl.bulky.html [hunk 1]
```diff
@@ -1 +1 @@
-<button class="btn btn-bulky<!-- BEGIN disabled --> ilSubmitInactive disabled<!-- END disabled --><!-- BEGIN engaged --> engaged<!-- END engaged -->"<!-- BEGIN with_data_action --> data-action="{ACTION}"<!-- END with_data_action --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id --> <!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --> <!-- BEGIN with_aria_pressed --> aria-pressed="{ARIA_PRESSED}"<!-- END with_aria_pressed --> >{ICON_OR_GLYPH}<div><span class="bulky-label">{LABEL}</span></div></button>
\ No newline at end of file
+<button class="btn btn-bulky<!-- BEGIN engaged --> engaged<!-- END engaged -->"<!-- BEGIN with_data_action --> data-action="{ACTION}"<!-- END with_data_action --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id --> <!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --> <!-- BEGIN with_aria_pressed --> aria-pressed="{ARIA_PRESSED}"<!-- END with_aria_pressed -->  <!-- BEGIN disabled --> disabled="true"<!-- END disabled -->>{ICON_OR_GLYPH}<div><span class="bulky-label">{LABEL}</span></div></button>
```
Comment (klees) [inline src/UI/templates/default/Button/tpl.bulky.html]:
```suggestion
<button class="btn btn-bulky<!-- BEGIN engaged --> engaged<!-- END engaged -->"<!-- BEGIN with_data_action --> data-action="{ACTION}"<!-- END with_data_action --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id --> <!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --> <!-- BEGIN with_aria_pressed --> aria-pressed="{ARIA_PRESSED}"<!-- END with_aria_pressed -->  <!-- BEGIN disabled --> disabled<!-- END disabled -->>{ICON_OR_GLYPH}<div><span class="bulky-label">{LABEL}</span></div></button>
```

File: src/UI/templates/default/Button/tpl.primary.html [hunk 2]
```diff
@@ -1 +1 @@
-<button class="btn btn-default btn-primary<!-- BEGIN disabled --> ilSubmitInactive disabled<!-- END disabled --><!-- BEGIN engaged --> engaged<!-- END engaged -->"<!-- BEGIN with_aria_pressed --> aria-pressed="{ARIA_PRESSED}"<!-- END with_aria_pressed --><!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --><!-- BEGIN with_data_action --> data-action="{ACTION}"<!-- END with_data_action --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id -->>{LABEL}</button>
\ No newline at end of file
+<button class="btn btn-default btn-primary<!-- BEGIN engaged --> engaged<!-- END engaged -->"<!-- BEGIN with_aria_pressed --> aria-pressed="{ARIA_PRESSED}"<!-- END with_aria_pressed --><!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --><!-- BEGIN with_data_action --> data-action="{ACTION}"<!-- END with_data_action --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id --><!-- BEGIN disabled --> disabled="true"<!-- END disabled -->>{LABEL}</button>
```
Comment (klees) [inline src/UI/templates/default/Button/tpl.primary.html]:
```suggestion
<button class="btn btn-default btn-primary<!-- BEGIN engaged --> engaged<!-- END engaged -->"<!-- BEGIN with_aria_pressed --> aria-pressed="{ARIA_PRESSED}"<!-- END with_aria_pressed --><!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --><!-- BEGIN with_data_action --> data-action="{ACTION}"<!-- END with_data_action --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id --><!-- BEGIN disabled --> disabled<!-- END disabled -->>{LABEL}</button>
```

File: src/UI/templates/default/Button/tpl.shy.html [hunk 3]
```diff
@@ -1 +1 @@
-<button class="btn btn-link<!-- BEGIN disabled --> ilSubmitInactive disabled<!-- END disabled -->"<!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --><!-- BEGIN with_data_action --> data-action="{ACTION}"<!-- END with_data_action --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id -->>{LABEL}</button>
+<button class="btn btn-link"<!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --><!-- BEGIN with_data_action --> data-action="{ACTION}"<!-- END with_data_action --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id --><!-- BEGIN disabled --> disabled="true"<!-- END disabled -->>{LABEL}</button>
```
Comment (klees) [inline src/UI/templates/default/Button/tpl.shy.html]:
```suggestion
<button class="btn btn-link"<!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --><!-- BEGIN with_data_action --> data-action="{ACTION}"<!-- END with_data_action --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id --><!-- BEGIN disabled --> disabled<!-- END disabled -->>{LABEL}</button>
```

File: src/UI/templates/default/Button/tpl.standard.html [hunk 4]
```diff
@@ -1 +1 @@
-<button class="btn btn-default<!-- BEGIN disabled --> ilSubmitInactive disabled<!-- END disabled --><!-- BEGIN engaged --> engaged<!-- END engaged -->" <!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --> <!-- BEGIN with_aria_pressed --> aria-pressed="{ARIA_PRESSED}"<!-- END with_aria_pressed --><!-- BEGIN with_data_action --> data-action="{ACTION}"<!-- END with_data_action --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id -->>{LABEL}</button>
\ No newline at end of file
+<button class="btn btn-default<!-- BEGIN engaged --> engaged<!-- END engaged -->" <!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --> <!-- BEGIN with_aria_pressed --> aria-pressed="{ARIA_PRESSED}"<!-- END with_aria_pressed --><!-- BEGIN with_data_action --> data-action="{ACTION}"<!-- END with_data_action --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id --><!-- BEGIN disabled --> disabled="true"<!-- END disabled -->>{LABEL}</button>
```
Comment (klees) [inline src/UI/templates/default/Button/tpl.standard.html]:
```suggestion
<button class="btn btn-default<!-- BEGIN engaged --> engaged<!-- END engaged -->" <!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --> <!-- BEGIN with_aria_pressed --> aria-pressed="{ARIA_PRESSED}"<!-- END with_aria_pressed --><!-- BEGIN with_data_action --> data-action="{ACTION}"<!-- END with_data_action --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id --><!-- BEGIN disabled --> disabled<!-- END disabled -->>{LABEL}</button>
```

File: src/UI/templates/default/Button/tpl.tag.html [hunk 5]
```diff
@@ -1 +1 @@
...
+<button class="btn btn-tag<!-- BEGIN rel_verylow --> btn-tag-relevance-verylow<!-- END rel_verylow --><!-- BEGIN rel_low --> btn-tag-relevance-low<!-- END rel_low --><!-- BEGIN rel_mid --> btn-tag-relevance-middle<!-- END rel_mid --><!-- BEGIN rel_high --> btn-tag-relevance-high<!-- END rel_high --><!-- BEGIN rel_veryhigh --> btn-tag-relevance-veryhigh<!-- END rel_veryhigh --><!-- BEGIN additional_classes --> {CLASSES}<!-- END additional_classes -->"<!-- BEGIN colors --> style="<!-- BEGIN bgcol -->background-color: {BGCOL}; <!-- END bgcol --><!-- BEGIN forecol -->color: {FORECOL};<!-- END forecol -->"<!-- END colors --><!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --><!-- BEGIN with_data_action --> data-action="{ACTION}"<!-- END with_data_action --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id --><!-- BEGIN disabled --> disabled="true"<!-- END disabled -->>{LABEL}</button>
```
Comment (klees) [inline src/UI/templates/default/Button/tpl.tag.html]:
```suggestion
<button class="btn btn-tag<!-- BEGIN rel_verylow --> btn-tag-relevance-verylow<!-- END rel_verylow --><!-- BEGIN rel_low --> btn-tag-relevance-low<!-- END rel_low --><!-- BEGIN rel_mid --> btn-tag-relevance-middle<!-- END rel_mid --><!-- BEGIN rel_high --> btn-tag-relevance-high<!-- END rel_high --><!-- BEGIN rel_veryhigh --> btn-tag-relevance-veryhigh<!-- END rel_veryhigh --><!-- BEGIN additional_classes --> {CLASSES}<!-- END additional_classes -->"<!-- BEGIN colors --> style="<!-- BEGIN bgcol -->background-color: {BGCOL}; <!-- END bgcol --><!-- BEGIN forecol -->color: {FORECOL};<!-- END forecol -->"<!-- END colors --><!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --><!-- BEGIN with_data_action --> data-action="{ACTION}"<!-- END with_data_action --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id --><!-- BEGIN disabled --> disabled<!-- END disabled -->>{LABEL}</button>
```

File: src/UI/templates/default/Button/tpl.toggle.html [hunk 6]
```diff
@@ -1,4 +1,4 @@
 <!-- BEGIN with_label --><label>{LABEL}</label><!-- END with_label -->
-<button class="il-toggle-button<!-- BEGIN on --> on<!-- END on --><!-- BEGIN disabled --> ilSubmitInactive disabled<!-- END disabled -->"<!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id --> aria-pressed="false">
+<button class="il-toggle-button<!-- BEGIN on --> on<!-- END on -->"<!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id --> aria-pressed="false"  <!-- BEGIN disabled --> disabled="true"<!-- END disabled -->>
```
Comment (klees) [inline src/UI/templates/default/Button/tpl.toggle.html]:
```suggestion
<button class="il-toggle-button<!-- BEGIN on --> on<!-- END on -->"<!-- BEGIN with_aria_label --> aria-label="{ARIA_LABEL}"<!-- END with_aria_label --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id --> aria-pressed="false"  <!-- BEGIN disabled --> disabled<!-- END disabled -->>
```

**Review by klees — APPROVED** [review]
Hi @Amstutz!

Thanks a lot for taking care of this. Kudos for streamlining the examples. One minor problem: [Boolean attributes in HTML might not work as expected](https://stackoverflow.com/questions/32745276/explicitly-set-disabled-false-in-the-html-does-not-work) so I suggested the according changes. Feel free to squash merge afterwards.

Best regards!

**Comment by klees** [comment]
This also once again shows, that we need a better strategy to check our html output.

**Comment by Amstutz** [comment]
As far as I know, disabled="true", disabled="disabled", disabled="foo", disabled="false" should all be equal. However, your proposal to just set disabled, results in invalid xml: "Specification mandate value for attribute disabled in Entity". Since I see that, disabled="true" might be somewhat confusing, I settled with disabled="disabled". 

I hope this is oke, Ill push the changes in a minute.

**Comment by klees** [comment]
Hi @Amstutz,

this is absolutely fine for me, and you are indeed correct about the various possibilities. I think the world still is a better place if we don't use `disabled="true"` because someone inevitably *will* think, that `disabled="false"` enables the element. Thanks for changing this. Sadly, the current HTML dialect is not an XML derivative because of boolean attributes without values (and some other reasons) anymore.

Best regards!

**Comment by Amstutz** [comment]
Oke, I implemented disabled="disabled" and pushed directly, got into some rebase issues, so I close this here.

**Comment by Amstutz** [comment]
See: https://github.com/ILIAS-eLearning/ILIAS/commit/40e6922815d3d8ebff48ff9cf1e6eceb6908aed3

**Pull request closed without merging**